app/src/pages/trace/SpanAnnotationForm.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useCallback, useRef } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { css } from "@emotion/react";
+
+import { Button, Flex, TextArea, TextField, View } from "@arizeai/components";
+
+export type CreateAnnotationInput = {
+  name: string;
+  score?: number | null;
+  label?: string | null;
+  explanation?: string | null;
+};
+
+export type SpanAnnotationFormProps = {
+  /**
+   * The initial data to populate the form with
+   */
+  initialData: CreateAnnotationInput;
+  /**
+   * Whether the form is read only
+   */
+  isReadOnly?: boolean;
+  /**
+   * Callback to call when the form is submitted
+   */
+  onSubmit?: (data: CreateAnnotationInput) => void;
+  /**
+   * Whether or not the form is being submitted
+   * @default false
+   */
+  isSubmitting?: boolean;
+};
+/**
+ * A form to create or edit a span annotation
+ */
+export function SpanAnnotationForm(props: SpanAnnotationFormProps) {
+  const { initialData, isReadOnly, onSubmit, isSubmitting = false } = props;
+  const formRef = useRef<HTMLFormElement>(null);
+  const {
+    control,
+    handleSubmit,
+    formState: { isDirty, isValid },
+    setError,
+  } = useForm({
+    defaultValues: initialData,
+    disabled: isReadOnly,
+  });
+
+  // Internal onSubmit that performs validation
+  const _onSubmit = useCallback(
+    (data: CreateAnnotationInput) => {
+      if (!data.label && (data.score as unknown as string) === "") {
```

Comment:
why is the check for these two different? score of "0" should still be okay with !data.score

Reply:
It's a an artifact of the input field producing strings when it's empty.

Reply:
ahh but in that case wouldn't the check for data.label and data.score just be the same?